File: 251001/Voytov/publisher/src/main/java/org/ex/distributed_computing/repository/NoticeRepository.java
package org.ex.distributed_computing.repository;

import java.util.List;
import org.ex.distributed_computing.model.Notice;
import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface NoticeRepository extends CrudRepository<Notice, Long> {

  List<Notice> findAll();
}

File: 251001/Voytov/discussion/src/main/kotlin/org/ex/discussion/repository/NoticeRepository.kt
package org.ex.discussion.repository

import org.ex.discussion.model.Notice
import org.springframework.data.cassandra.repository.Query
import org.springframework.data.repository.CrudRepository
import org.springframework.data.repository.query.Param
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface NoticeRepository : CrudRepository<Notice, Long> {

    @Query(
        value = "SELECT * FROM notice WHERE id = :id ALLOW FILTERING;",
        allowFiltering = true
    )
    override fun findById(@Param("id") id: Long): Optional<Notice>

    @Query(
        value = "DELETE FROM notice WHERE id = :id AND country = :country;"
    )
    fun delete(@Param("id") id: Long, @Param("country") country: String)
}